### ISSUE TITLE
Character Limit Not Being Implemented Where Needed

### ISSUE THREAD

**Issue by KELSEYDOWLING7** [episphere/questionnaire#347]
In looking at the address responses for Module 4 for Geocoding, we noticed that there's a decently high proportion of people responding with both street number and street name in the street number address prompts. We noticed that the markup here does not match the data dictionary; the markup allows up to 99999 characters whereas the Data Dictionary limits it to 6.

`[D_121490150?] What is the full street address of the place where you live now? Street number |__|id=D_255248624 min=0 max=999999| Full Street name |__|id=D_945532934| Apartment, suite, unit, building, etc. |__|id=D_469838242| City |__|id=D_303500597| State/Province |__|id=D_195068098| Zip code |__|id=D_202784871| Country |__|id=D_831127170|`

![image](https://github.com/episphere/questionnaire/assets/101136647/ea329956-5e06-49d8-8b83-c155b2f6a0ae)

In issue #227 Tony mentioned the need for minlen / maxlen to properly limit the number of characters participants can put into these types of text prompts where we might possibly expect a combination of number and letter (houses like 12 or condos/apartments like 12B). This should be applied to every address street number, including work places. 

We may need to revisit the minlen/maxlen as opposed to min/max for all text boxes where a character limit is set but not enforced as well. We have the character limits in the Module QC, so once that's implemented on the module data we can find these other instances. Unless anyone can think of any important ones offhand

**Comment by KELSEYDOWLING7** [comment]
Bringing this issue to the top of everyone's inboxes to discuss on today's call

**Comment by m-j-horner** [comment]
@boyd-mj @joshid-ims @huip-ims 

Deepti- Please update Module 4 word document and the Markdown with the minlen/maxlen= 6 so that the fields are concordant with what is in the data dictionary. 

Variables are **street** **number** for residential address, seasonal address, childhood address, work address, school address.

**Comment by joshid-ims** [comment]
Thanks! I assume this is for English and Spanish.

**Comment by m-j-horner** [comment]
@joshid-ims  Correct. The minlen/maxlen applies to English and Spanish for the October release

**Comment by joshid-ims** [comment]
Just to make sure, minlen=0 and maxlen=6, right?

**Comment by joshid-ims** [comment]
Thanks!

**Comment by woodruffr** [comment]
The English M4 Questionnaire with this fix was delivered today.

**Comment by woodruffr** [comment]
I note that Aileen has a spreadsheet to update the Notes section for minlen=0 maxlen=6. It was changed in the questionnaires and also in the mark-ups.

**Comment by cunnaneaq** [comment]
Looks fixed in dev to me this is for CHILDADD

![Image](https://github.com/user-attachments/assets/1f661af4-3ed2-4c2c-abe6-ab41eb0bd59b)

**Comment by cunnaneaq** [comment]
Same for HOMEADD

**Comment by cunnaneaq** [comment]
Also fixed for SEASADD, CURWORK, PREWORK
